tests/eas/load_tracking.py

```diff
@@ -0,0 +1,225 @@
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from bart.common.Utils import select_window, area_under_curve
+from devlib.utils.misc import memoized
+from trappy.stats.grammar import Parser
+import pandas as pd
+
+from test import LisaTest, experiment_test
+
+UTIL_SCALE = 1024
+# Time in seconds to allow for util_avg to converge (i.e. ignored time)
+UTIL_AVG_CONVERGENCE_TIME = 0.3
+# Allowed margin between expected and observed util_avg value
+ERROR_MARGIN_PCT = 15
+
+class FreqInvarianceTest(LisaTest):
+    """
+    Goal
+    ====
+    Basic check for frequency invariant load tracking
+
+    Detailed Description
+    ====================
+    This test runs the same workload on the most capable CPU on the system at a
+    cross section of available frequencies. The trace is then examined to find
+    the average activation length of the workload, which is combined with the
+    known period to estimate an expected mean value for util_avg for each
+    frequency. The util_avg value is extracted from scheduler trace events and
+    its mean is compared with the expected value (ignoring the first 300ms so
+    that the signal can stabilize). The test fails if the observed mean is
+    beyond a certain error margin from the expected one. load_avg is then
+    similarly compared with the expected util_avg mean, under the assumption
+    that load_avg should equal util_avg when system load is light.
+
+    Expected Behaviour
+    ==================
+    Load tracking signals are scaled so that the workload results in roughly the
+    same util & load values regardless of frequency.
+    """
+
+    test_conf = {
+        'tools'    : [ 'rt-app' ],
+        'ftrace' : {
+            'events' : [
+                'sched_switch',
+                'sched_load_avg_task',
+                'sched_load_avg_cpu',
+                'sched_pelt_se',
+            ],
+        },
+        'modules': ['cpufreq', 'cgroups'],
+    }
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(FreqInvarianceTest, cls).init(*args, **kwargs)
+
+    @memoized
+    @classmethod
+    def _get_cpu(cls, target):
+        # Run on a 'big' CPU, or any CPU if not big.LITTLE
+        if hasattr(target, 'bl'):
+            return target.bl.bigs[0]
+        else:
+            return 0
+
+    @classmethod
+    def _getExperimentsConf(cls, test_env):
+        cpu = cls._get_cpu(test_env.target)
+
+        # 10% periodic RTApp workload:
+        wloads = {
+            'fie_10pct' : {
+                'type' : 'rt-app',
+                'conf' : {
+                    'class' : 'periodic',
+                    'params' : {
+                        'duty_cycle_pct': 10,
+                        'duration_s': 3,
```

Comment:
Could this be reduce to 1 or max 2 [s]... it should be more than enough for such a short period task and it could save some time while running multiple tests.